### PR TITLE
Kuba openfaas sync call

### PR DESCRIPTION
**Pull request by kubatyszko**
Adding a way to invoke openfaas functions synchronously, waiting for the invocation to complete.

---

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/openfaas/hooks/openfaas.py [hunk 1]
```diff
@@ -68,16 +69,32 @@ def deploy_function(self, overwrite_function_if_exist: bool, body: Dict[str, Any
                 self.log.info("Function deployed %s", self.function_name)
 
     def invoke_async_function(self, body: Dict[str, Any]) -> None:
-        """Invoking function"""
+        """Invoking function asynchronously"""
         url = self.get_conn().host + self.INVOKE_ASYNC_FUNCTION + self.function_name
-        self.log.info("Invoking function %s", url)
+        self.log.info("Invoking function asynchronously %s", url)
         response = requests.post(url, body)
```
Comment (potiuk) [inline airflow/providers/openfaas/hooks/openfaas.py:75]: Should we use `json=body` consistently ? Right now it looks like they might encode the data differently (`json=body` will be json-encoded and `body` will be form-encoded). Was this  deliberate distinction? https://docs.openfaas.com/reference/async/ does not seem to indicate there is any difference.

**Review by potiuk — COMMENTED** [review]
Looks good, just one nit/question with post encoding

**Comment by potiuk** [comment]
Also there are some static code failures :). I recommend using `pre-commit`

File: airflow/providers/openfaas/hooks/openfaas.py (hunk 1 above)
Comment (kubatyszko) [reply airflow/providers/openfaas/hooks/openfaas.py:75]: I've corrected it.

**Comment by potiuk** [comment]
We've fixed failing master (the failing tests were due to that). Fixing the static checks and rebasing shoudl make it green :)

**Comment by kubatyszko** [comment]
@potiuk static checks succeeded finally, some other (seemingly unrelated) checks are failing.

**Comment by kubatyszko** [comment]
Hmm, something wrong with mock_requests. It doesn't even work for the async openfaas call (which I did not modify):

`
3170
  >       raise exceptions.NoMockAddress(request)
3171
  E       requests_mock.exceptions.NoMockAddress: No mock address: POST http://open-faas.io/async-function/function_name
3172
  
3173
  /usr/local/lib/python3.6/site-packages/requests_mock/adapter.py:258: NoMockAddress`

**Comment by kubatyszko** [comment]
Ok, I think I got it as far as I can, now it's failing with rabbitMQ error, assuming unrelated to my code:

`===============================================================================================
609
               Checking integrations and backends
610
  ===============================================================================================
611
  -----------------------------------------------------------------------------------------------
612
  Kerberos: .OK.  
613
  MongoDB: OK.  
614
  Redis: OK.  
615
  RabbitMQ: ....................ERROR: Maximum number of retries while checking service. Exiting 
616
  Service could not be started!
617
  
618
  $ nc -zvv rabbitmq 5672
619
  DNS fwd/rev mismatch: rabbitmq != docker-compose_rabbitmq_1.docker-compose_default
620
  rabbitmq [172.19.0.7] 5672 (amqp) : Connection refused
621
   sent 0, rcvd 0
622
 `

**Comment by potiuk** [comment]
It's a transient error - it happens sometimes. But as such - it can be merged I think. One last review.

**Review by potiuk — APPROVED** [review]

**Comment by kubatyszko** [comment]
Woohoo, thank you ! (and as of 2 minutes ago I'm finally in 2021 ;) )

**Comment by potiuk** [comment]
:tada: :tada: :tada: :tada: :tada: :tada: :tada:

**Comment by potiuk** [comment]
Nothing to look for . Not much different than 2020 here :)

**Pull request merged**